packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -257,6 +282,7 @@ export default {
         }
       );
 
+      res.results = uniqBy('id', concat(res.results, resWithIds.results));
       ctx.body = res;
     } else {
       const result = await strapi.entityService.load(model, { id }, targetField, queryParams);
```

Comment:
shouldn't we also apply the permission query when loading a XToOne relation?

Reply:
yes indeed I probably missed this one